src/app/TopicDashboard/ActiveResourceContainer.js

```diff
@@ -1,41 +1,104 @@
+    newOrder.splice(destination.index, 0, draggableId);
+    this.props.dispatch(updateRescOrder(newOrder, this.props.parentId));
   };
 
   render() {
-    const { resources, resourceOrder } = this.props;
-
+    const { resources, rescOrder } = this.props;
     return (
-      <section className="active-resources-container">
-        <ul className="active-resources-list">
-          {resourceOrder.map(rescID => {
-            if (
-              resourceOrder &&
-              resources &&
-              resourceOrder.length > 0 &&
-              resources[rescID]
-            ) {
-              return resources[rescID].completed === false ? (
-                <li key={rescID} className="resource-item-container">
-                  <ResourceItem resource={resources[rescID]} />
-                </li>
-              ) : null;
-            } else {
-              return null;
-            }
-          })}
-          <AddResourceForm />
-        </ul>
-      </section>
+      <DragDropContext onDragEnd={this.onDragEnd}>
+        <section className="active-resources-container">
+          <button
+            // className="add-resource-button"
+            type="button"
+            onClick={this.handleScrollClick}
+          >
+            Add Resource
+          </button>
+          <Droppable droppableId="droppable-1">
+            {provided => (
+              <ul ref={provided.innerRef} className="active-resources-list">
+                {rescOrder.map((rescID, index) => {
+                  if (
+                    rescOrder &&
+                    resources &&
+                    rescOrder.length > 0 &&
+                    resources[rescID]
+                  ) {
+                    return resources[rescID].completed === false ? (
+                      <Draggable
+                        key={rescID}
+                        draggableId={rescID}
+                        index={index}
+                      >
+                        {provided => {
+                          return (
+                            <>
+                              <li
+                                ref={provided.innerRef}
+                                {...provided.draggableProps}
+                                {...provided.dragHandleProps}
+                                className="resource-item-container"
+                              >
+                                <ResourceItem resource={resources[rescID]} />
+                              </li>
+                              {provided.placeholder}
+                            </>
+                          );
+                        }}
+                      </Draggable>
+                    ) : null;
+                  } else {
+                    return null;
+                  }
+                })}
+                {provided.placeholder}
+              </ul>
+            )}
+          </Droppable>
+
+          <AddResourceForm id="123" ref1={this.Form1} />
```

Comment:
Can we give this ID and ref more meaningful names?

Reply:
done :)